comp/dogstatsd/server/parse.go

```diff
@@ -238,8 +248,27 @@ func (p *parser) parseFloat64List(rawFloats []byte) ([]float64, error) {
 }
 
 // extractContainerID parses the value of the container ID field.
+// If the field is prefixed by `in-`, it corresponds to the cgroup controller's inode of the source
```

Comment:
This change broadens the scope of the "parser" to include resolving the container field into a container ID as well.
I can see how this is a convenient place to implement this, but this does put the inode-resolution in the parsing hot-path of the dogstatsd server, rather than deferring the entity lookup to the tagger as I would have expected.

The container id from here is used (eventually) in https://github.com/DataDog/datadog-agent/blob/9f01a78cdc4c6f81fd0d9e598f35cae8f3f01cb8/pkg/tagger/global.go#L271 

Would it be a big ask to move this logic into the tagger and create a separate `Entity` type (eg, current [container entity](https://github.com/DataDog/datadog-agent/blob/74bd32ae85fde53213e30b3f9f7f1a32bef56ddd/pkg/util/containers/entity.go#L20)) to represent the unresolved inode, that way we can treat this as an opaque set of bytes from the dogstatsd parsing perspective? 

This would keep a better separation of concerns, but also would complicate this change a bit.

Reply:
Seen together, we can keep the current implementation for now and come back to it later. A Jira card has been created.